source/plugins/go/src/utils.go

```diff
@@ -119,17 +126,71 @@ func CreateMDSDClient() {
 	  FluentSocketPath:"/var/run/mdsd/default_fluent.socket",
 	  WriteTimeout: 5 * time.Second,
 	  RequestAck: true}) */
+	mdsdfluentSocket := "/var/run/mdsd/default_fluent.socket"	
+	if containerType != "" && strings.Compare(strings.ToLower(containerType), "prometheussidecar") == 0 {
+	   mdsdfluentSocket = fmt.Sprintf("/var/run/mdsd-%s/default_fluent.socket", containerType)
+	} 
 	conn, err := net.DialTimeout("unix",
-		"/var/run/mdsd/default_fluent.socket", 10*time.Second)
+		mdsdfluentSocket, 10*time.Second)
 	if err != nil {
 		Log("Error::mdsd::Unable to open MDSD msgp socket connection %s", err.Error())
 		//log.Fatalf("Unable to open MDSD msgp socket connection %s", err.Error())
 	} else {
-		Log("Successfully created MDSD msgp socket connection")
+		Log("Successfully created MDSD msgp socket connection: %s", mdsdfluentSocket)
 		MdsdMsgpUnixSocketClient = conn
 	}
 }
 
+//mdsdSocketClient to write msgp messages for KubeMonAgent Events
+func CreateMDSDClientKubeMon(containerType string) {
+	if MdsdKubeMonMsgpUnixSocketClient != nil {
+		MdsdKubeMonMsgpUnixSocketClient.Close()
+		MdsdKubeMonMsgpUnixSocketClient = nil
+	}
+	/*conn, err := fluent.New(fluent.Config{FluentNetwork:"unix",
+	  FluentSocketPath:"/var/run/mdsd/default_fluent.socket",
+	  WriteTimeout: 5 * time.Second,
+	  RequestAck: true}) */
+	mdsdfluentSocket := "/var/run/mdsd/default_fluent.socket"	
```

Comment:
why is there a client creation code per data type ?

Reply:
The reason separate client per data type is to so these clients can be concurrent and indenpendly to talk to socket endpoint.  client creation creates separate client for that data type.

Reply:
refactored the code to reuse same create method.